content/python/concepts/strings/terms/zfill/zfill.md

```diff
@@ -0,0 +1,91 @@
+---
+Title: '.zfill()' 
+Description: 'Takes an int and returns a copy of the string with a number of zeros preppended as a suffix , i.e "no. of zeros" + string.'
+Subjects:
+  - 'Data Science'
+  - 'Computer Science'
+Tags:
+  - 'Strings'
+  - 'Methods'
+  - 'Functions'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/analyze-data-with-python'
+---
+
+The **`.zfill()`**  method pads a string with zeros on the left to maintain a specific length. It takes an integer argument, and the number of zeros added is determined by the difference between the specified length and the length of the original string.
+
+> **Note:** The `.zfill()` method does not change the string it is used on.
+If no argument is passed into the method. A type error is thrown
+
+## Syntax
+```pseudo
+string.zfill(arg(int))
+```
+
+## Codebyte Example
+The following example shows how `.zfill()` returns a copy of that string with leading zeros. 
+
+```py
+first_string = "Codecademy"
+second_string = "@Codecademy"
+third_string = "cat"
+
+print(first_string)
+print(second_string)
+
+# parameter is less than/ equal to string length, output will be the same string.
+print(first_string.zfill(10))
+print(second_string.zfill(10))
+
+print(first_string.zfill(12))
+print(second_string.zfill(12))
+print(third_string.zfill(12))
+```
+This would output:
+
+```shell
+Codecademy
+@Codecademy
+
+Codecademy
+@Codecademy
+
+00Codecademy
+0@Codecademy
+000000000cat
+
+```
+
+## Example
```

Comment:
This section is really not necessary. All functions with required parameters will throw an error if the parameter is missing. It isn't something specific to `.zfill()`